pulser-simulation/pulser_simulation/hamiltonian.py

```diff
@@ -0,0 +1,638 @@
+            self._build_basis_and_op_matrices()
+        self._build_collapse_operators(cfg)
+        self._config = cfg
+        if not (
+            "SPAM" in self.config.noise_types
+            and self.config.state_prep_error > 0
+        ):
+            self._bad_atoms = {qid: False for qid in self._qid_index}
+        if "doppler" not in self.config.noise_types:
+            self._doppler_detune = {qid: 0.0 for qid in self._qid_index}
+        # Noise, samples and Hamiltonian update routine
+        self._construct_hamiltonian()
+
+    def add_config(self, config: NoiseModel) -> None:
+        """Updates the current configuration with parameters of another one.
+
+        Mostly useful when dealing with multiple noise types in different
+        configurations and wanting to merge these configurations together.
+        Adds simulation parameters to noises that weren't available in the
+        former NoiseModel. Noises specified in both NoiseModels will keep
+        former noise parameters.
+
+        Args:
+            config: NoiseModel to retrieve parameters from.
+        """
+        if not isinstance(config, NoiseModel):
+            raise ValueError(f"Object {config} is not a valid `NoiseModel`")
+
+        not_supported = (
+            set(config.noise_types) - SUPPORTED_NOISES[self._interaction]
+        )
+        if not_supported:
+            raise NotImplementedError(
+                f"Interaction mode '{self._interaction}' does not support "
+                f"simulation of noise types: {', '.join(not_supported)}."
+            )
+
+        old_noise_set = set(self.config.noise_types)
+        new_noise_set = old_noise_set.union(config.noise_types)
+        diff_noise_set = new_noise_set - old_noise_set
+        print(diff_noise_set)
+        # Create temporary param_dict to add noise parameters:
+        param_dict: dict[str, Any] = asdict(self._config)
+        # Begin populating with added noise parameters:
+        param_dict["noise_types"] = tuple(new_noise_set)
+        if "SPAM" in diff_noise_set:
+            param_dict["state_prep_error"] = config.state_prep_error
+            param_dict["p_false_pos"] = config.p_false_pos
+            param_dict["p_false_neg"] = config.p_false_neg
+        if "doppler" in diff_noise_set:
+            param_dict["temperature"] = config.temperature
+        if "amplitude" in diff_noise_set:
+            param_dict["laser_waist"] = config.laser_waist
+        if "dephasing" in diff_noise_set:
+            param_dict["dephasing_prob"] = config.dephasing_prob
+        if "depolarizing" in diff_noise_set:
+            param_dict["depolarizing_prob"] = config.depolarizing_prob
+        if "eff_noise" in diff_noise_set:
+            param_dict["eff_noise_opers"] = config.eff_noise_opers
+            param_dict["eff_noise_probs"] = config.eff_noise_probs
+        # update runs:
+        param_dict["runs"] = config.runs
+        param_dict["samples_per_run"] = config.samples_per_run
```

Comment:
I don't see the point of having this complexity here, I think it should stay in `QutipEmulator` where it originally was. Plus, 'runs' and 'samples_per_run' are not part of the noise model

Since Hamiltonian is an internal class, I think we can just skip this method altogether and set a new config from `QutipEmulator` instead

Reply:
I agree I was not a fan of having add_config in Hamiltonian

Reply:
Hum, however, I see `runs` and `samples_per_run` as being attributes of the NoiseModel class... https://github.com/pasqal-io/Pulser/blob/develop/pulser-core/pulser/backend/noise_model.py#L79

Reply:
Yes, I forgot they were there, I probably confused them with `quitp_options`, sorry